src/diffusion_for_multi_scale_molecular_dynamics/models/score_networks/mlp_score_network.py

```diff
@@ -91,17 +120,32 @@ def _forward_unchecked(
                 Defaults to False.
 
         Returns:
-            computed_scores : the scores computed by the model.
+            computed_scores : the scores computed by the model in an AXL namedtuple.
         """
-        relative_coordinates = batch[NOISY_RELATIVE_COORDINATES]
+        relative_coordinates = batch[NOISY_AXL].X
         # shape [batch_size, number_of_atoms, spatial_dimension]
 
         sigmas = batch[NOISE].to(relative_coordinates.device)  # shape [batch_size, 1]
         noise_embedding = self.noise_embedding_layer(
             sigmas
-        )  # shape [batch_size, embedding_dimension]
+        )  # shape [batch_size, noise_embedding_dimension]
 
-        input = torch.cat([self.flatten(relative_coordinates), noise_embedding], dim=1)
+        atom_types = batch[NOISY_AXL].A
+        atom_types_one_hot = torch.nn.functional.one_hot(
```

Comment:
Use the method `class_index_to_onehot`  instead.

Also, define a variable for `num_classes`.